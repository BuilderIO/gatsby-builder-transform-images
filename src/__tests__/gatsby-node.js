jest.mock(`gatsby-source-filesystem`);
import {
  createResolvers as pluginCreateResolvers,
  setFieldsOnGraphQLNodeType
} from "../gatsby-node";
import expect from "expect";

describe("createResolvers", () => {
  it("should define a resolver for the localFile field in SanityImageAsset type", () => {
    const createResolvers = jest.fn();

    pluginCreateResolvers({ createResolvers, actions: {} });
    const argumentsFirstCall = createResolvers.mock.calls[0][0];

    expect(argumentsFirstCall.SanityImageAsset).not.toBeFalsy();
    expect(argumentsFirstCall.SanityImageAsset.localFile).not.toBeFalsy();

    const { url } = argumentsFirstCall.SanityImageAsset.localFile.resolve(
      { url: "randon-url" },
      { width: 30, height: 20, fit: "crop", format: "jpg" }
    );

    expect(url).toBe("randon-url?w=30&fm=jpg&h=20&fit=crop");
  });

  it("should extend the SanityImageAsset type adding a localFile field of type File", () => {
    const extendedSanityImageAsset = setFieldsOnGraphQLNodeType({
      type: { name: "SanityImageAsset" }
    });

    expect(extendedSanityImageAsset.localFile.type).toBe("File");
    expect(extendedSanityImageAsset.localFile.args).toMatchSnapshot();
  });
});
