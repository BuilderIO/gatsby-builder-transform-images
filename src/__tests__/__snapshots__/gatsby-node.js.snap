// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createResolvers should extend the SanityImageAsset type adding a localFile field of type File 1`] = `
Object {
  "fit": Object {
    "defaultValue": "crop",
    "type": Object {},
  },
  "format": Object {
    "type": Object {
      "name": "SanityTransformImagesImageFormat",
      "values": Object {
        "JPG": Object {
          "value": "jpg",
        },
        "PJPG": Object {
          "value": "pjpg",
        },
        "PNG": Object {
          "value": "png",
        },
        "WEBP": Object {
          "value": "webp",
        },
      },
    },
  },
  "height": Object {
    "type": Object {},
  },
  "width": Object {
    "type": Object {},
  },
}
`;
